### PR TITLE
Fix CI failures related to conversion of np.floating to dtype

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

This PR prevent the conversion using `np.floating` as it is [deprecated](https://numpy.org/devdocs/release/1.19.0-notes.html#converting-certain-types-to-dtypes-is-deprecated) in numpy 1.19.

It fixes CI failures in #4730

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by sciunto** [comment]
looks like it does not fully fixes the CI :)

File: skimage/util/dtype.py [hunk 1]
```diff
@@ -244,6 +244,8 @@ def _convert(image, dtype, force_copy=False, uniform=False):
         if force_copy:
             image = image.copy()
         return image
+    else:
+        dtype = np.float64
```
Comment (lagru) [inline skimage/util/dtype.py:248]: Did you mean `dtypeobj_out`? I think `dtype` is never used from this point onwards...

Comment (rfezzani) [reply skimage/util/dtype.py:248]: Good catch, it should be `dtype_out` ;)

**Comment by rfezzani** [comment]
Ci failures are fixed by #4735

**Comment by jni** [comment]
To clarify my position: this PR currently takes NumPy's deprecated behaviour and hardcodes it into skimage. Instead, we should follow NumPy's guidance here. Users doing `_convert(arr, np.floating)` should get the warning (although for some reason I couldn't trigger it locally), users using `img_as_float` should get the fixed (future) behaviour.

**Comment by jni** [comment]
fwiw, I *do* think we should consider making `img_as_float` default to `np.float` rather than `np.float64`. But that can be resolved another day, this matches current behaviour.

**Comment by sciunto** [comment]
@scikit-image/core Do we plan to release a minor version to include this fix?

**Comment by jni** [comment]
Looks like it was backported to the 0.17.x branch, so I think we could release 0.17.3 with this. @emmanuelle could you handle that? I'd be available today and/or Thursday to help if needed.

**Pull request merged**